drivers/sensor/bmi160/bmi160_spi.c

```diff
@@ -0,0 +1,80 @@
+/* Bosch BMI160 inertial measurement unit driver
+ *
+ * Copyright (c) 2020 Linumiz
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://ae-bst.resource.bosch.com/media/_tech/media/datasheets/BST-BMI160-DS000.pdf
+ */
+
+#include <init.h>
+#include <drivers/sensor.h>
+#include <sys/byteorder.h>
+#include <kernel.h>
+#include <sys/__assert.h>
+#include <logging/log.h>
+
+#include "bmi160.h"
+
+#define BMI160_CHIP_ID		0xD1 /* Constant for bmi160 sensor */
+#define DT_DRV_COMPAT		bosch_bmi160
+
+LOG_MODULE_REGISTER(BMI160, CONFIG_SENSOR_LOG_LEVEL);
+
+int bmi160_init(struct device *dev)
+{
+	struct bmi160_device_data *bmi160 = dev->driver_data;
+	int ret;
+
+	bmi160->spi = device_get_binding(bmi160->bus_name);
+	if (!bmi160->spi) {
+		LOG_DBG("SPI master controller not found: %s.",
+			    bmi160->bus_name);
+		return -EINVAL;
+	}
+
+	bmi160->spi_cfg.operation = SPI_WORD_SET(8);
+	bmi160->spi_cfg.frequency = DT_INST_PROP(0, spi_max_frequency);
```

Comment:
This makes impossible to have >1 instance defined in DT. This driver already uses a generic initializer, so it can be stored in the instance config structure.

Reply:
oops, my mistake